src/app/debug/table/table.component.ts

```diff
@@ -272,30 +294,29 @@ export class TableComponent implements OnInit, OnDestroy {
     this.filterService.setCurrentRecords(this.tableSettings.uniqueValues);
   }
 
-  toggleCheck(report: any, event: MouseEvent): void {
+  toggleCheck(report: Report, event: MouseEvent): void {
     event.stopPropagation();
     report.checked = !report.checked;
-    if (this.allRowsSelected && !report.checked) {
-      this.allRowsSelected = false;
+    if (report.checked) {
+      this.amountOfSelectedReports++;
+    } else {
+      this.amountOfSelectedReports--;
     }
-    this.allRowsSelected = this.checkIfAllRowsSelected();
   }
 
-  checkIfAllRowsSelected(): boolean {
-    for (let reportMetadata of this.tableSettings.reportMetadata) {
-      if (!reportMetadata.checked) {
-        return false;
-      }
+  toggleSelectAll(): void {
+    if (this.amountOfSelectedReports === this.tableSettings.reportMetadata.length) {
+      this.setCheckedForAllReports(false);
+      this.amountOfSelectedReports = 0;
+    } else {
+      this.setCheckedForAllReports(true);
+      this.amountOfSelectedReports = this.tableSettings.reportMetadata.length;
     }
-    return true;
   }
```

Comment:
It might be a good idea to change the checkboxes to the angular way using mat-checkbox. That way we can just grab the length of the SelectionModel. These methods can be simplified this way.

[example of mat-checkbox](https://stackblitz.com/edit/angular-mat-table-checkbox-select-all-ivt1dw)

Reply:
I wanted to use the mat-checkbox, but I saw it had it's own styling with preset themes. So if you can tell me how to custom style it, I will gladly implement it.

Reply:
[stackoverflow explanation](https://stackoverflow.com/questions/51984326/how-to-style-a-mat-checkbox-properly)

TL;DR
"To style the non checked box:"
```css
    .mat-checkbox-background{
    background-color: transparent;
    svg path{
        fill: transparent;
    }
    .mat-checkbox-checkmark-path {
         stroke:transparent;
         //stroke is the checked mark only
        }
    }
```

"To style the checked box:"
```css
.mat-checkbox-indeterminate.mat-accent .mat-checkbox-background, .mat-checkbox-checked.mat-accent .mat-checkbox-background {
    background-color: #2cc3f8 !important;
    svg path {
        fill: #2cc3f8 !important;
    }
}
```

Reply:
I think these two methods are not complex enough to use a new component of Angular Material that our project depends on